queries.sql

```diff
@@ -15,3 +15,58 @@ SELECT * FROM animals WHERE neutered =  true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg BETWEEN  10.4 and 17.3;
+
+-- transaction start
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022, 01, 01';
+
+SAVEPOINT save_1;
+
+UPDATE animals SET weight_kg = weight_kg*-1;
+
+ROLLBACK TO save_1;
+
+UPDATE animals SET weight_kg = weight_kg*-1 WHERE weight_kg < 0;
+
+COMMIT;
+
+-- transaction end
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT
+     neutered,
+     COUNT(*) AS total_animals,
+     SUM(escape_attempts) AS total_escapes
+ FROM
+     animals
+ GROUP BY
+     neutered;
+
+SELECT
+     neutered,
+     COUNT(*) AS total_animals,
+     SUM(escape_attempts) AS total_escapes
+ FROM
+     animals
+ GROUP BY
+     neutered;
+
+SELECT
+    species,
+    MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+ FROM
+    animals
+ GROUP BY
+    species;
+
+SELECT
+    AVG(escape_attempts) AS average_escape_attempts
+ FROM
+    animals
+ WHERE
+    date_of_birth BETWEEN '1990, 01, 01' AND '2000, 12, 31';
```

Comment:
- [ Required ] I think to answer the question _What is the average number of escape attempts per animal type of those born between 1990 and 2000?_  you would have to `Group by species` the result set that is returned by the `WHERE` clause. Please see the code suggestion below for an idea of how to achieve this.

## Example code
```SQL
   SELECT species, AVG(escape_attempts) FROM animals
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31'
   GROUP BY species;
```